mkosi/distributions/centos.py

```diff
@@ -132,148 +134,142 @@ def gpgurls() -> tuple[str, ...]:
         )
 
     @classmethod
-    def repository_variants(cls, config: MkosiConfig, repo: str) -> list[Repo]:
+    @listify
```

Comment:
Do we even need this everywhere? Why not have all these return generators and pass an `Iterable` to `setup_dnf()`? Or just convert the entire thing to a list just before we invoke `setup_dnf()`. Adding the listify seems like noise primarily

Reply:
I added this because you wrote:
> But I would prefer we keep it simple and just return a list of repos like we do everywhere else

I'll rework it to be a generator.